### PR TITLE
[16.0][MIG] sale_order_product_assortment: Migration to version 16.0

### PR DESCRIPTION
**Pull request by carolinafernandez-tecnativa**
- Redo migration v16 https://github.com/OCA/sale-workflow/pull/2712 as this one is not valid due to the changes that makes it very slow.
- Change Setup to SetupClass tests

@Tecnativa
TT44355
@pedrobaeza @victoralmau

### CODE REVIEW

File: sale_order_product_assortment/tests/test_sale_order_product_assortment.py [hunk 1]
```diff
@@ -1,17 +1,19 @@
 # Copyright 2020 Tecnativa - Carlos Roca
+# Copyright 2024 Tecnativa - Carolina Fernandez
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
 from odoo.tests.common import TransactionCase
 
 
 class TestProductAssortment(TransactionCase):
```
Comment (pedrobaeza) [inline sale_order_product_assortment/tests/test_sale_order_product_assortment.py]: Speed up a bit the execution of the test with:

```
from odoo.odoo.addons.base.tests.common import BaseCommon
```

and inheriting the class from that one.

**Review by pedrobaeza — APPROVED** [review]

**Review by Deriman-Alonso — APPROVED** [review]
LGTM 👍, code review.

**Pull request merged**